src/poetry/core/factory.py

```diff
@@ -78,6 +78,8 @@ def _add_package_group_dependencies(
         group: str | DependencyGroup,
         dependencies: DependencyConfig,
     ) -> None:
+        from poetry.core.packages.dependency_group import MAIN_GROUP
```

Comment:
We should do a try catch here and on ImprotError set `MAIN_GROUP="default"` until we bump core.

Reply:
Interesting idea. I don't think it will do any good here in `poetry-core`. But I can do this in the other two PRs.

Reply:
Ah wrong PR. I meant to comment on the poetry one.